python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -53,33 +66,54 @@ def __init__(self):
         ret_val_doc += "For the 'update_blocked' status, data will contain a reason key containing an explanation."
 
         self.parameters = {"return_value": {"description": ret_val_doc, "type": "dict" }}
-            
+
+    def _parse_arguments(self, parameters):
+        """
+        Parses the list of arguments from the command line.
+
+        :param parameters: The content of argv that hasn't been processed by the tank command.
+
+        :returns: The core version. None if --version wasn't specified.
+        """
+        parser = TkOptParse()
```

Comment:
Can we be agile? I don't think it needs to be super complicated.

We have several commands that take *optional* flags. Quick grep gave me `--version=xxx`, `--preserve-yaml`, `--backout`, `--full`, `--symlink`, `--force`, `--no-storage-check`, `--all`, `--filter=xxx`, `--external=xxxx`. Just supporting that would be a really great useful first step.

This is always a pre-flight thing, we look for certain flags, they are always optional, we parse, remove args we discover from the from the args string and then return it back to the main tank command for further processing.

Sorry if i am missing something!